.gitlab/kernel_version_testing/system_probe.yml

```diff
@@ -299,37 +293,47 @@ kernel_matrix_testing_setup_env:
     - cat $CI_PROJECT_DIR/stack.outputs
     - pulumi logout
   after_script:
     - export AWS_PROFILE=agent-qa-ci
     - !reference [.shared_filters_and_queries]
-    - mkdir -p $CI_PROJECT_DIR/libvirt/log/x86_64 $CI_PROJECT_DIR/libvirt/log/arm64
-    - mkdir -p $CI_PROJECT_DIR/libvirt/xml/x86_64 $CI_PROJECT_DIR/libvirt/xml/arm64
-    - INSTANCE_TYPE=m5d.metal
-    - !reference [.get_instance_ip_by_type]
-    - X64_INSTANCE_IP=$INSTANCE_IP
-    - ssh -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$X64_INSTANCE_IP" "sudo virsh list --name | grep -v -E '^$' | xargs -I '{}' sh -c \"sudo virsh dumpxml '{}' > /tmp/ddvm-xml-'{}'.txt\""
-    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$X64_INSTANCE_IP:/tmp/ddvm-*.log" $CI_PROJECT_DIR/libvirt/log/x86_64/
-    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$X64_INSTANCE_IP:/tmp/ddvm-xml-*" $CI_PROJECT_DIR/libvirt/xml/x86_64
-    - INSTANCE_TYPE=m6gd.metal
+    - mkdir -p $CI_PROJECT_DIR/libvirt/log/$ARCH $CI_PROJECT_DIR/libvirt/xml/$ARCH
     - !reference [.get_instance_ip_by_type]
-    - ARM64_INSTANCE_IP=$INSTANCE_IP
-    - ssh -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$ARM64_INSTANCE_IP" "sudo virsh list --name | grep -v -E '^$' | xargs -I '{}' sh -c \"sudo virsh dumpxml '{}' > /tmp/ddvm-xml-'{}'.txt\""
-    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$ARM64_INSTANCE_IP:/tmp/ddvm-*.log" $CI_PROJECT_DIR/libvirt/log/arm64/
-    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$ARM64_INSTANCE_IP:/tmp/ddvm-xml-*" $CI_PROJECT_DIR/libvirt/xml/arm64
+    - ssh -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP" "sudo virsh list --name | grep -v -E '^$' | xargs -I '{}' sh -c \"sudo virsh dumpxml '{}' > /tmp/ddvm-xml-'{}'.txt\""
+    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP:/tmp/ddvm-*.log" $CI_PROJECT_DIR/libvirt/log/$ARCH
+    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP:/tmp/ddvm-xml-*" $CI_PROJECT_DIR/libvirt/xml/$ARCH
   artifacts:
     when: always
     paths:
-      - $LibvirtSSHKeyX86
-      - $LibvirtSSHKeyX86.pub
-      - $LibvirtSSHKeyARM
-      - $LibvirtSSHKeyARM.pub
-      - $STACK_DIR
       - $CI_PROJECT_DIR/stack.outputs
       - $CI_PROJECT_DIR/libvirt
 
+kernel_matrix_testing_setup_env_arm64:
+  extends:
+    - .kernel_matrix_testing_setup_env
+  variables:
+    EC2_ARM_INSTANCE_TYPE: "m6gd.metal"
+    INSTANCE_TYPE: $EC2_ARM_INSTANCE_TYPE
+    INSTANCE_TYPE_ARG: "--instance-type-arm=$INSTANCE_TYPE"
+    ARCH: arm64
+    AMI_ID_ARG: "--arm-ami-id=$KERNEL_MATRIX_TESTING_ARM_AMI_ID"
+    LibvirtSSHKey: $CI_PROJECT_DIR/libvirt_rsa-arm
+    VMCONFIG_PATH: $CI_PROJECT_DIR/test/new-e2e/system-probe/config/vmconfig-arm64.json
+
+kernel_matrix_testing_setup_env_x64:
+  extends:
+    - .kernel_matrix_testing_setup_env
+  variables:
+    EC2_X86_INSTANCE_TYPE: "m5d.metal"
```

Comment:
Do we still needs `EC2_X86_INSTANCE_TYPE` and `INSTANCE_TYPE`, or only `INSTANCE_TYPE_ARG`?